tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_optimizer.py

```diff
@@ -169,27 +168,38 @@ def add_slot(var, slot_name, initializer="zeros"):
                                           dtype=var.dtype)
       else:
         initial_value = initializer
-      strategy = distribute_ctx.get_strategy()
-      with strategy.extended.colocate_vars_with(var):
-        if isinstance(var, de.TrainableWrapper):
-          weight = de.create_slots(var, initial_value, slot_name,
-                                   var._shared_name, self._bp_v2)
-        else:
-          weight = variables.Variable(
-              name="%s/%s" % (
-                  var._shared_name,
-                  slot_name,
-              ),  # pylint: disable=protected-access
-              dtype=var.dtype,
-              trainable=False,
-              initial_value=initial_value,
-          )
-      backend.track_variable(weight)
-      slot_dict[slot_name] = weight
-      self._restore_slot_variable(slot_name=slot_name,
-                                  variable=var,
-                                  slot_variable=weight)
-      self._weights.append(weight)
+      with self._distribution_strategy_scope():
```

Comment:
Please format all the files firstly, refer to: https://github.com/tensorflow/recommenders-addons/blob/master/STYLE_GUIDE.md